### PR TITLE
provide alternative merging option

### PR DESCRIPTION
**Pull request by imorland**
image

### CODE REVIEW

File: extend.php [hunk 1]
```diff
@@ -53,4 +56,7 @@
 
     (new Extend\Middleware('forum'))
         ->insertBefore(HandleErrors::class, Middleware\Redirection::class),
+
+    // (new Extend\ApiController(ListPostsController::class))
```
Comment (davwheat) [inline extend.php]: We can remove this, right?

File: js/package.json [hunk 2]
```diff
@@ -6,9 +6,9 @@
   "dependencies": {
     "@flarum/prettier-config": "^1.0.0",
     "flarum-tsconfig": "^1.0.2",
-    "flarum-webpack-config": "^1.0.0",
-    "webpack": "^4.46.0",
-    "webpack-cli": "^4.9.1"
+    "flarum-webpack-config": "^2.0.0",
```
Comment (davwheat) [inline js/package.json:9]: We'll need to bump core's constraint to 1.2+

File: js/src/forum/components/DiscussionMergeModal.js [hunk 3]
```diff
@@ -48,19 +52,67 @@ export default class DiscussionMergeModal extends Modal {
...
+        `type_${key}`,
+        <div>
+          <input type="radio" id={`type_${key}`} checked={this.type() === key} onclick={this.changeType.bind(this, key)} />
+          &nbsp;
+          <label htmlFor={`type_${key}`}>{app.translator.trans(`fof-merge-discussions.forum.modal.type_${key}_label`)}</label>
+        </div>,
+        priority
+      );
+
+      priority = priority - 5;
```
Comment (davwheat) [inline js/src/forum/components/DiscussionMergeModal.js]:
```suggestion
      priority -= 5;
```

File: js/src/forum/components/DiscussionMergeModal.js [hunk 4]
```diff
@@ -48,19 +52,67 @@ export default class DiscussionMergeModal extends Modal {
...
+          <input type="radio" id={`ordering_${key}`} checked={this.order() === key} onclick={this.changeOrdering.bind(this, key)} />
+          &nbsp;
+          <label htmlFor={`ordering_${key}`}>{app.translator.trans(`fof-merge-discussions.forum.modal.ordering_${key}_label`)}</label>
+          &nbsp;
+          <Tooltip text={app.translator.trans(`fof-merge-discussions.forum.modal.ordering_${key}_help`)}>{icon('fas fa-info-circle')}</Tooltip>
+        </div>,
+        priority
+      );
+
+      priority = priority - 5;
```
Comment (davwheat) [inline js/src/forum/components/DiscussionMergeModal.js]:
```suggestion
      priority -= 5;
```

**Review by davwheat — CHANGES_REQUESTED** [review]
I can't really review the PHP that well, but it all seems fine to me.

Just a few comments.

File: js/package.json (hunk 2 above)
Comment (imorland) [reply js/package.json:9]: Already present in `master` :)

File: extend.php (hunk 1 above)
Comment (imorland) [reply extend.php]: 🙈 probably... lemme check

File: js/package.json (hunk 2 above)
Comment (davwheat) [reply js/package.json:9]: Oop, my bad

**Pull request merged**